homeworks/class3/routes/cats.js

```diff
@@ -0,0 +1,91 @@
+var catData = require('../cat-creator');
+var Cat = require('mongoose').model('Cat');
+
+module.exports.showAll = function (req, res) {
+	var pageData = {
+		'page-title' : "All",
+		'page-header': "All Cats",
+		'page-subheader': "Sorted by Age"
+	};
+
+	// Get Cats from DB
+	Cat.find()
+	   .sort({age: 1})
+	   .exec( function(err, cats) {
+			if (err) {
+				pageData.error = "Failed to locate cats!";
+				res.render('listing', pageData);
+				return console.error("Problem finding cats: " + cats, err);
+			} else {
+				pageData.data = cats;
+				res.render('listing', pageData);
+			}
+	});
+};
+
+module.exports.sortColor = function (req, res) {
+	var pageData = {
+		'page-title' : "Color Filter",
+		'page-header': "Cats",
+		'page-subheader': "Filtered by Color"
+	};
+
+	// Get Color in URI
+	var segments = req.url.split('/');
```

Comment:
Yup. This is relevant to my comment above about using `/:color` instead of `/*`, which allows you to access the value by `req.params.color`, rather than having to roll your own string hack.